tasks/ModelTask.cpp

```diff
@@ -4,127 +4,138 @@
+        joints_in.elements.push_back( base::JointState::Effort(0.0) );
+        joints_out.names.push_back( (*joint)->GetName() );
+        joints_out.elements.push_back( base::JointState::Position(0.0) );
+    }
 }
-//======================================================================================
-void ModelTask::setLinkPorts()
+
+void ModelTask::setupLinks(NameVector link_names)
 {
-    // Get all links from a model and creates a Rock component InputPort
-	links = model->GetLinks();
-	for(Link_V::iterator link = links.begin(); link != links.end(); ++link)
+    // Create Rock output ports for the links defined in the configuration file.
+	gazebo_links = model->GetLinks();
+	for(Link_V::iterator link = gazebo_links.begin(); link != gazebo_links.end(); ++link)
 	{
-		gzmsg << "RockBridge: found link: " << world->GetName() + "/" + model->GetName() + 
-				"/" + (*link)->GetName() << std::endl;
-		gzmsg << "RockBridge: create link InputPort in Rock." << std::endl;
-		link_port = new RTT::InputPort<base::Vector3d>( "_link_" + (*link)->GetName() + "_in");
-		ports()->addPort( *link_port );
-		link_port_list.push_back( std::make_pair(link_port,*link) );
+        for(NameVector::iterator it = link_names.begin(); it != link_names.end(); ++it)
+        {
+            if( (*it).compare( (*link)->GetName() ) == 0 )
+            {
+                gzmsg << "RockBridge: found link: " << world->GetName() + "/" + model->GetName() +
+                    "/" + (*link)->GetName() << std::endl;
+
+                // Create the ports dynamicaly
+                RBSOutPort* link_out_port = new RBSOutPort( "link_"+ (*link)->GetName() +"_samples" );
+                ports()->addPort(*link_out_port);
+                link_list.push_back( std::make_pair(link_out_port,*link) );
+            }
+        }
 	}
 }
-//======================================================================================
+
 void ModelTask::updateHook()
 {
     updateJoints();
     updateLinks();
 }
-//======================================================================================
+
 void ModelTask::updateJoints()
 {
-	for(JointPort_V::iterator it = joint_port_list.begin();it != joint_port_list.end(); ++it)
-	{
-		double effort = 0;	
-		(*it).first->readNewest(effort);
-		if(effort != RTT::NoData)
-		{
-			// Define the limits for the joint effort
-			try{
-			    if((-1.0 <= effort) && (effort <= 1.0)){
-			        (*it).second->SetForce(0,effort);
-			    }else{
-			        throw;
-			    }
-			}catch(...){
-			    // Joint effort value out of range 
-			    gzmsg << "RockBridge: error - effort value out of range (-1 <= effort <= 1)." << std::endl;
-			}
-		}
-	}
+    _joints_cmd.readNewest( joints_in );
+    for(Joint_V::iterator it = gazebo_joints.begin(); it != gazebo_joints.end(); ++it )
+    {
+        // Apply effort to joint
+        double effort = joints_in.getElementByName( (*it)->GetName() ).effort;
+        if( ! isnan(effort) )
```

Comment:
Do you mean hasEffort() ?!
I cannot access the dfki link above, I don't have permission.